### PR TITLE
dhall-lsp-server: Fix hovering in presence of nested lets

### LINKED ISSUE #1510: dhall-lsp-server may show the wrong type on hover
This type should actually be `Bool -> { home : Text, name : Text }`:

https://github.com/dhall-lang/dhall-haskell/blob/1b46f1885f25c16e369b3f455c3950e4837ac49f/dhall-lsp-server/tests/Main.hs#L40

There may be more bugs – ideally all the type information logic should have test coverage:

https://github.com/dhall-lang/dhall-haskell/blob/cc1814be9858f62e664b0d2114e711fd4431bd2c/dhall-lsp-server/src/Dhall/LSP/Backend/Typing.hs#L31-L72

### PR DESCRIPTION
**Pull request by EggBaconAndSpam**
Fixes #1510 by reintroducing a helper that wraps 'naked' multilets in Src annotations.

### CODE REVIEW

**Comment by EggBaconAndSpam** [comment]
> Thanks! slightly_smiling_face
> 
> You'll need to update the golden test to reflect the new (correct) result!

Yep!

File: dhall-lsp-server/src/Dhall/LSP/Backend/Parsing.hs [hunk 1]
```diff
@@ -73,7 +83,7 @@ getLetIdentifier src@(Src left _ text) =
   where parseLetIdentifier = do
           setSourcePos left
           _let
-          whitespace
+          nonemptyWhitespace
```
Comment (sjakobi) [inline dhall-lsp-server/src/Dhall/LSP/Backend/Parsing.hs]: Why do we actually have all this duplicated parsing logic? Is it impossible to share this with `dhall`'s parser? Could you maybe add some documentation for this @EggBaconAndSpam?

Comment (EggBaconAndSpam) [reply dhall-lsp-server/src/Dhall/LSP/Backend/Parsing.hs]: I believe the problem was that the parser is parameterised by the parser for import expressions, which means we can't easily export its components. I'll have to think about it. I admit that the current situation is a bit of a hack.

Comment (Gabriel439) [reply dhall-lsp-server/src/Dhall/LSP/Backend/Parsing.hs]: @EggBaconAndSpam: You can import export additional parsers if you need to by reusing the same trick we use for exporting the parser for `importExpression`:

https://github.com/dhall-lang/dhall-haskell/blob/53320c8e1944c9c8d72c380fe1232185aa03b276/dhall/src/Dhall/Parser/Expression.hs#L105-L125

Comment (EggBaconAndSpam) [reply dhall-lsp-server/src/Dhall/LSP/Backend/Parsing.hs]: I think I'll look into that separately. It seems there are a few more bugs that recently crept in that I want to look at.

**Comment by EggBaconAndSpam** [comment]
I'll merge this now and refactor the parsing part separately.

**Pull request merged**